diff --git a/scripts/irods/controller.py b/scripts/irods/controller.py
index d3d46c613..3de3525d8 100644
--- a/scripts/irods/controller.py
+++ b/scripts/irods/controller.py
@@ -223,8 +223,16 @@ class IrodsController(object):
         l = logging.getLogger(__name__)
         for attempt in range(retry_count):
             l.debug('Waiting for iRODS server to shut down. Attempt #%s', attempt)
-            if self.get_server_pid() is None:
+            server_pid = self.get_server_pid()
+            if server_pid is None:
                 return
+
+            # checking for zombieness
+            proc = psutil.Process(server_pid)
+            if proc.status() == psutil.STATUS_ZOMBIE:
+                l.info('Zombie found, ignoring...')
+                return
+
             time.sleep(1)
         raise IrodsError('iRODS server failed to shut down.')
